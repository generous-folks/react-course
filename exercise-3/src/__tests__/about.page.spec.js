import React from 'react';
import { Link } from 'react-router-dom';

import { shallow } from 'enzyme';

import { AboutPage } from '../pages/about.page';

const getWrapper = () => shallow(<AboutPage />);

let wrapper;

describe('<AboutPage />', () => {
  beforeEach(() => {
    jest.clearAllMocks();

    wrapper = getWrapper();
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should contain Link', () => {
    expect(wrapper.find(Link).prop('to')).toBe('/');
  });

  it('should contain h2', () => {
    expect(wrapper.find('h2').exists()).toBeTruthy();
  });
});
