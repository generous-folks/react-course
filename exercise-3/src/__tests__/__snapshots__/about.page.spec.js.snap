// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AboutPage /> should render correctly 1`] = `
<div>
  <h2>
    About
  </h2>
  <Link
    to="/"
  >
    Return to Home
  </Link>
</div>
`;
